colossalai/fx/profiler/profiler.py

```diff
@@ -27,33 +28,104 @@ def is_autogradable(x):
     return isinstance(x, torch.Tensor) and x.is_floating_point()
 
 
-# super-dainiu:
-# x.detach() will change the unique identifier of data_ptr
-# we need to handle this in a stupid way
-def detach(x):
+def detach_variables(x):
     if isinstance(x, torch.Tensor):
         requires_grad = x.requires_grad
-        x.requires_grad_(False)
-        x.requires_grad_(requires_grad)
+        x = x.detach()
+        x.requires_grad = requires_grad
```

Comment:
wait, this is incorrect for `MetaInfoProp`, please recover the comment and the original function